aten/src/ATen/native/cuda/LossCTC.cu

```diff
@@ -739,6 +740,9 @@ Tensor ctc_loss_backward_gpu_template(const Tensor& grad_out, const Tensor& log_
     C10_CUDA_KERNEL_LAUNCH_CHECK();
   }
 
+  if (!is_batched) {
+    return grad.squeeze(1);
+  }
```

Comment:
Honestly I'm still a bit confused about the changes to this file - if we handle things above this at the `ctc_loss` level, I wouldn't expect to need changes here. Can you help me understand this a bit more?